base/src/java/lang/util.d

```diff
@@ -387,31 +387,103 @@ version(Tango){
 
     class Format{
         template UnTypedef(T) {
-            version(Tango){
-                mixin(r"static if (is(T U == typedef))
-                    alias std.traits.Unqual!(U) UnTypedef;
-                else
-                    alias std.traits.Unqual!(T) UnTypedef;");
-            } else { // Phobos
-                alias std.traits.Unqual!(T) UnTypedef;
-            }
+            alias UnTypedef = std.traits.Unqual!(T);
         }
-        static String opCall(A...)( String _fmt, A _args ){
-            //Formatting a typedef is deprecated
+        static String opCall(A...)(in String fmt, A _args) {
+            auto app = std.array.appender!(String)();
             std.typetuple.staticMap!(UnTypedef, A) args;
-            foreach(i, _a; _args) {
-                version(Tango){
-                    mixin(r"static if (is(T U == typedef))
-                        args[i] = cast(U) _a;
-                    else
-                        args[i] = _a;");
-                } else { // Phobos
-                    args[i] = _a;
+            int argIndex = 0;
+            bool increment = false; // should the argIndex be incremented
+
+            // 0 = "basic type", 1 = array, 2 = assocArray (no impl. yet)
+            ubyte[] isRange = new ubyte[_args.length];

```

Comment:
`_args.length` is known at compile time, so you can use a static array instead. It should be more efficient.

Reply:
Fixed.